src/test/java/org/terracotta/statistics/simulation/latency/GraphUtils.java

```diff
@@ -0,0 +1,144 @@
+      }
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  public static void writeGraph(File output,
+                                List<LongSample> samples, Map<String, Percentile> percentiles, DomainOrder domainOrder,
+                                String xAxisLabel, String yAxisLabel, Dimension dimension,
+                                DoubleUnaryOperator xAxisTransform, DoubleUnaryOperator yAxisTransform) {
+    String title = String.format("count: %s, min: %s, max: %s%n%s",
+        samples.size(),
+        percentiles.get("min"),
+        percentiles.get("max"),
+        stringify(percentiles));
+    LongSampleDataset dataset = new LongSampleDataset(samples, domainOrder, xAxisTransform, yAxisTransform);
+    JFreeChart chart = ChartFactory.createXYLineChart(title, xAxisLabel, yAxisLabel, dataset, PlotOrientation.VERTICAL, true, false, false);
+    XYPlot plot = (XYPlot) chart.getPlot();
+    plot.setFixedLegendItems(new LegendItemCollection());
+    addMarkers(percentiles, plot);
+    writeChart(output, chart, dimension);
+  }
+
+  public static void writeGraph(File output,
+                                List<Histogram.Bucket> buckets, Map<String, Percentile> percentiles, DomainOrder domainOrder,
+                                String xAxisLabel, Dimension dimension,
+                                DoubleUnaryOperator xAxisTransform) {
+    String title = String.format("count: %s, min: %s, max: %s%n%s",
+        (long) buckets.stream().mapToDouble(Histogram.Bucket::count).sum(),
+        percentiles.get("min"),
+        percentiles.get("max"),
+        stringify(percentiles));
+    HistogramDataset dataset = new HistogramDataset("", buckets, domainOrder, xAxisTransform);
+    JFreeChart chart = ChartFactory.createXYBarChart(title, xAxisLabel, false, "height", dataset, PlotOrientation.VERTICAL, true, false, false);
+    XYPlot plot = (XYPlot) chart.getPlot();
+    plot.setFixedLegendItems(new LegendItemCollection());
+    XYBarRenderer renderer = (XYBarRenderer) plot.getRenderer();
+    renderer.setDrawBarOutline(true);
+    writeChart(output, chart, dimension);
+  }
+
+  private static void addMarkers(Map<String, Percentile> percentiles, XYPlot plot) {
+    for (Map.Entry<String, Percentile> entry : percentiles.entrySet()) {
+      ValueMarker marker = new ValueMarker(entry.getValue().value());
+      marker.setPaint(Color.black);
+      marker.setLabel(entry.getKey() + ": " + entry.getValue().toString(FORMATER));
+      marker.setLabelFont(new Font("SansSerif", Font.BOLD, 12));
+      marker.setLabelOffset(new RectangleInsets(3.0, 5, 5.0, 3.0));
+      marker.setLabelTextAnchor(TextAnchor.BOTTOM_LEFT);
+      plot.addRangeMarker(marker);
+    }
+  }
+
+  public static void writeChart(File output, JFreeChart chart, Dimension dimension) {
+    try {
+      ChartUtils.saveChartAsPNG(output, chart, dimension.width, dimension.height);
+    } catch (RuntimeException e) {
```

Comment:
No because if it throws a IO it is OK, but if it throws another exception like Illegal Argument or illegal state or index out of bounds, or something else, then an empty file is created...